feat: Add Amazon Textract as document loader

schadem opened the pull request:
Description: Adding support for [Amazon Textract](https://aws.amazon.com/textract/) as a PDF document loader



Please make sure you're PR is passing linting and testing before submitting. Run `make format`, `make lint` and `make test` to check this locally. -> checked and worked fine

schadem committed:
add Amazon Textract as document loader

schadem committed:
add Amazon Textract as document loader

brettdbrewer committed:
updated to use new llm_util query (#8591)

- Description: added memgraph_graph.py which defines the MemgraphGraph
class, subclassing off the existing Neo4jGraph class. This lets you
query the Memgraph graph database using natural language. It leverages
the Neo4j drivers and the bolt protocol.
- Dependencies: since it is a subclass off of Neo4jGraph, it is
dependent on it and the GraphCypherQA Chain implementations. It is
dependent on the Neo4j drivers being present. It is dependent on having
a running Memgraph instance to connect to.
  - Tag maintainer: @baskaryan
  - Twitter handle: @villageideate
- example usage can be seen in this repo
https://github.com/brettdbrewer/MemgraphGraph/

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

baskaryan committed:
mv (#8595)

MthwRobinson committed:
docs: update `unstructured` install instructions (#8596)

### Summary

Updates the `unstructured` install instructions. For
`unstructured>=0.9.0`, dependencies are broken out by document type and
the base `unstructured` package includes fewer dependencies. `pip
install "unstructured[local-inference]"` has been replace by `pip
install "unstructured[all-docs]"`, though the `local-inference` extra is
still supported for the time being.

### Reviewers

- @rlancemartin
- @eyurtsev
- @hwchase17

sfc-gh-jcarroll committed:
Add StreamlitChatMessageHistory (#8497)

Add a StreamlitChatMessageHistory class that stores chat messages in
[Streamlit's Session
State](https://docs.streamlit.io/library/api-reference/session-state).

Note: The integration test uses a currently-experimental Streamlit
testing framework to simulate the execution of a Streamlit app. Marking
this PR as draft until I confirm with the Streamlit team that we're
comfortable supporting it.

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

schadem committed:
fix linting and format

naveentatikonda committed:
[OpenSearch] : Fix AOSS Initialization (#8600)

### Description
This PR fixes the AOSS Initialization in Opensearch.

### Maintainers
@rlancemartin, @eyurtsev, @navneet1v

Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

wnmurphy committed:
Add firestore_client param to FirestoreChatMessageHistory if caller already has one; also lets them specify GCP project, etc. (#8601)

Existing implementation requires that you install `firebase-admin`
package, and prevents you from using an existing Firestore client
instance if available.

This adds optional `firestore_client` param to
`FirestoreChatMessageHistory`, so users can just use their existing
client/settings. If not passed, existing logic executes to initialize a
`firestore_client`.

---------

Co-authored-by: Bagatur <baskaryan@gmail.com>

baskaryan committed:
normalized openai embeddings embed_query (#8604)

we weren't normalizing when embedding queries

3coins commented:
@schadem 
Thanks for submitting the PR. You might need to upload the poetry lock file to fix the CI error.
```
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

Because langchain depends on amazon-textract-caller (<2) which doesn't match any versions, version solving failed.
```

rjanardhan3 committed:
Fireworks integration (#8322)

Description - Integrates Fireworks within Langchain LLMs to allow users
to use Fireworks models with Langchain, mainly for summarization.

Issue - Not applicable
Dependencies - None
Tag maintainer - @rlancemartin

---------

Co-authored-by: Raj Janardhan <rajjanardhan@Rajs-Laptop.attlocal.net>

hwchase17 committed:
cast to string when measuring token length (#8617)

naveentatikonda committed:
[Opensearch] : Fix the service validation in http_auth (#8609)

### Description
OpenSearch supports validation using both Master Credentials (Username
and password) and IAM. For Master Credentials users will not pass the
argument `service` in `http_auth` and the existing code will break. To
fix this, I have updated the condition to check if service attribute is
present in http_auth before accessing it.

### Maintainers
@baskaryan @navneet1v

Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

Comendeiro committed:
Add local support for audio models (PR #7329) (#7591)

- Description: run the poetry dependencies
  - Issue: #7329
  - Dependencies: any dependencies required for this change,
  - Tag maintainer: @rlancemartin

---------

Co-authored-by: Harrison Chase <hw.chase.17@gmail.com>
Co-authored-by: Bagatur <baskaryan@gmail.com>

schadem commented on libs/langchain/langchain/document_loaders/pdf.py:
```diff
@@ -100,6 +108,32 @@ def _is_valid_url(url: str) -> bool:
+        except ValueError:
+            return False
+
+    @staticmethod
+    def _load_s3_file(file_path: str, temp_pdf: str) -> None:
```
Thx. 1, 2, 3 and also 4 are done.
It is inconsistent with the http downloader and has duplication for the temp path code, which I would like to refactore in a separate ticket. We could also keep the file in memory instead of downloading locally.

baskaryan committed:
bump 250 (#8632)

eyurtsev committed:
Re-add __add__ functionality for messages (revert #8245) (#8489)

This PR reverts #8245, so `__add__` is defined on base messages.

Resolves issue: https://github.com/langchain-ai/langchain/issues/8472

schadem committed:
fix comments

schadem commented on libs/langchain/langchain/document_loaders/pdf.py:
```diff
@@ -440,3 +474,108 @@ def load(self) -> List[Document]:
+
+    """
+
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. \
+            Please install it with `pip install amazon-textract-caller` \
+            or add to your requirements.txt."
+        )
```
I reference the package in the parameters for the constructor, in the constructor would be too late.

Commit:
Enterprise support form in airtable (#8607)

kacperlukawski committed:
Refactor Qdrant integration (#8634)

This small PR introduces new parameters into Qdrant (`on_disk`), fixes
some tests and changes the error message to be more clear.

Tagging: @baskaryan, @rlancemartin, @eyurtsev

schadem committed:
fix final lint
